runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp

```diff
@@ -392,15 +392,24 @@ void LightningSimulator::PartialCounts(DataView<double, 1> &eigvals, DataView<in
     }
 }
 
-auto LightningSimulator::Measure(QubitIdType wire) -> Result
+auto LightningSimulator::Measure(QubitIdType wire, int8_t postselect) -> Result
 {
+    // Check validity of postselect value
+    RT_FAIL_IF(postselect < -1 || postselect > 1, "Invalid postselect value");
+    bool enabledPostselect = postselect == -1 ? false : true;
+
     // get a measurement
     std::vector<QubitIdType> wires = {reinterpret_cast<QubitIdType>(wire)};
 
     std::vector<double> probs(1U << wires.size());
     DataView<double, 1> buffer_view(probs);
     this->PartialProbs(buffer_view, wires);
 
+    // Check validity of probability for postselect value
+    if (enabledPostselect) {
+        RT_FAIL_IF(probs[postselect] == 0, "Probability of postselect value is 0");
+    }
+
     std::random_device rd;
```

Comment:
Is postselect supported here, or is it a TODO addressed in a separate PR?

Reply:
It is supported, just not yet implemented. But it will be addressed in this PR

Reply:
I think it is done now :)